src/SerialCommandHandler.cpp

```diff
@@ -54,17 +49,17 @@ std::vector<std::string> SerialCommandHandler::sendCommand(std::string cmd, cons
       throw ChimeraTK::runtime_error(err);
     }
     outputStrVec.push_back(readStr);
-  } // end for
+  }
 
   return outputStrVec;
-} // end sendCommand
+}
 
 /**********************************************************************************************************************/
 
 std::string SerialCommandHandler::waitAndReadline() const {
   auto readData = _serialPort->readline();
   if(not readData.has_value()) {
-    throw std::logic_error("FIXME: BAD INTERFACE");
+    throw ChimeraTK::logic_error("FIXME: BAD INTERFACE");
```

Comment:
Ah, this actually has to stay a std::logic_error.

The specification exactly defines where ChimeraTK::logic_errors can occur. They always can be prevented by checking a pre-condition in the public API. As this is the abstract API of the Device class which does not know about this particular backend, this is not possible. Plus ChimeraTK::logic_errors are not allowed in this context.

Maybe we should print "Sue the programmer, he fucked up hard" and call std::terminate() here instead of throwing. There is design/implementation missing, which I wanted to express with "FIXME: BAD INTERFACE". 

I guess we have to address this. I created a ticket https://redmine.msktools.desy.de/issues/13739

Please revert for this branch.